### DISCUSSION TITLE
Supabase Auth-UI: No way to know the current displayed view

### DISCUSSION THREAD

**Discussion by Artyom22** [supabase/supabase#14646] [Feature Requests]
### Is your feature request related to a problem? Please describe.
Currently, if I'm on the "Sign in" page and, for example, I click on the "Forgot your password?", I have no way of knowing that I was "navigated" to another display. It will be helpful to have this capability to display additional messages or components on the page based on the current type of view, add logging, hide/display components, and so on.

### Describe the solution you'd like
Add a possibility to pass a callback that will receive the current view as an argument and will be triggered whenever the view is changed.

**Comment by justin-gurtz** [comment]
I was about to create a similar discussion, I have a use case where I need to know the Auth UI `view` state in order to update surrounding components.

(By the way this seems similar to https://github.com/orgs/supabase/discussions/13219)

**Reply by Artyom22** [reply]
there is an open PR for this feature [https://github.com/supabase/auth-ui/pull/183](https://github.com/supabase/auth-ui/pull/183), but it hasn't been moved forward for a few months

**Reply by justin-gurtz** [reply]
Ah nice, didn't see that. I could really use that functionality...
